cs/common/Bag.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace VW
+{
+    public interface IBag<T>
+    {
+        void Add(T item);
+
+        T Remove();
+
+        IEnumerable<T> RemoveAll();
+    }
+
+    public static class Bag
+    {
+        public static IBag<T> Synchronized<T>(IBag<T> bag)
```

Comment:
Looked at ConcurrentBag. Doesn’t have exactly the interface I’m looking for. Also there is no base class/interface that I could switch in/out if thread safety isn’t needed…

From: Luong Hoang [mailto:notifications@github.com]
Sent: Wednesday, May 4, 2016 4:45 PM
To: JohnLangford/vowpal_wabbit <vowpal_wabbit@noreply.github.com>
Cc: Markus Cozowicz <marcozo@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [JohnLangford/vowpal_wabbit] Thread-safe CLR label parsing (#986)


In cs/common/Bag.cs<https://github.com/JohnLangford/vowpal_wabbit/pull/986#discussion_r62111401>:

> +using System.Threading.Tasks;

> +

> +namespace VW

> +{

> +    public interface IBag<T>

> +    {

> +        void Add(T item);

> +

> +        T Remove();

> +

> +        IEnumerable<T> RemoveAll();

> +    }

> +

> +    public static class Bag

> +    {

> +        public static IBag<T> Synchronized<T>(IBag<T> bag)

Is ConcurrentBag<> not what you want here?

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/JohnLangford/vowpal_wabbit/pull/986/files/cd4a90714ba1f4cbd82a460b99641f3aacc2158e#r62111401>